### PR TITLE
Add streaming support for netcore and netfx

### PR DESCRIPTION
**Pull request by apoorvdeshmukh**
This PR adds support for streaming JSON datatype to and from SQL Server.

### CODE REVIEW

File: src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/JsonTest/JsonStreamTest.cs [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+            {
+                using (StreamReader jsonFile = File.OpenText(jsonFilename))
+                {
+                    cmd.Parameters.Add("@jsondata", Microsoft.Data.SqlDbTypeExtensions.Json, -1).Value = jsonFile;
+                    cmd.ExecuteNonQuery();
+                }
+            }
+        }
+
+        private void deleteFile(string filename)
```
Comment (saurabh500) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/JsonTest/JsonStreamTest.cs:111]: All C# functions start with Upper case letters

**Comment by deepaksa1** [comment]
Does the test cases cover `SqlDataReader.GetStream()` ?

**Comment by apoorvdeshmukh** [comment]
> Does the test cases cover `SqlDataReader.GetStream()` ?

As per the [docs](https://learn.microsoft.com/en-us/dotnet/api/microsoft.data.sqlclient.sqldatareader.getstream?view=sqlclient-dotnet-standard-5.2) GetStream() API is only supported for binary, image, varbinary and udt datatypes hence not required as part of the current testing.

File: src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlEnums.cs [hunk 2]
```diff
@@ -124,6 +124,7 @@ private static bool _IsCharType(SqlDbType type) =>
             type == SqlDbType.Char ||
             type == SqlDbType.VarChar ||
             type == SqlDbType.Text ||
+            type == SqlDbTypeExtensions.Json ||
```
Comment (deepaksa1) [inline src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlEnums.cs:127]: Is the usual read operation working as expected after adding this? (We had a problem earlier in read which was fixed by removing this)

Comment (apoorvdeshmukh) [reply src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlEnums.cs:127]: Yes. The read issue was caused due to the existing code path trying to find the encoding and collation for char datatypes. The fix for the same is already in place for read/write changes.

**Review by deepaksa1 — APPROVED** [review]
LGTM

**Review by saurabh500 — APPROVED** [review]

**Pull request merged**